commands/build.go

```diff
@@ -96,89 +68,85 @@ func runBuild(dockerCli command.Cli, in buildOptions) (err error) {
 		end(err)
 	}()
 
-	noCache := false
-	if in.noCache != nil {
-		noCache = *in.noCache
-	}
-	pull := false
-	if in.pull != nil {
-		pull = *in.pull
-	}
+	_, err = runBuildContext(ctx, dockerCli, in.BuildOptions, os.Stdin, in.progress, in.invoke != "", nil)
```

Comment:
If my reading of the code here is right, we only support `--invoke` when going through the `runBuildOnBuilder` path? That seems fine to me, especially as it looks we'll make this the default eventually?

If so, I think we might be able to remove the `allowNoOutput` flag at this point.

Reply:
Thank you for the review. Fixed.